minecode/visitors/generic.py

```diff
@@ -77,3 +76,86 @@ def process_request(purl_str):
 
     if error_msg:
         return error_msg
+
+
+def map_directory_listed_package(package_url):
+    """
+    Add a directory listed `package_url` to the PackageDB.
+
+    Return an error string if any errors are encountered during the process
+    """
+    from minecode.model_utils import add_package_to_scan_queue
+    from minecode.model_utils import merge_or_create_package
+
+    packages = [p for p in info(str(package_url)) or []]
+
+    if not packages:
+        error = f"Could not find package using fetchcode: {package_url}"
+        logger.error(error)
+        return error
+
+    package_data = packages[0].to_dict()
+
+    # Remove obsolete Package fields see https://github.com/nexB/fetchcode/issues/108
+    del package_data["api_url"]
+    del package_data["license_expression"]
+    del package_data["declared_license"]
+    del package_data["root_path"]
+    del package_data["contains_source_code"]
+    del package_data["purl"]
+
+    package = PackageData(**package_data)
+
+    db_package, _, _, error = merge_or_create_package(package, visit_level=0)
+
+    # Submit package for scanning
+    if db_package:
+        add_package_to_scan_queue(db_package)
+
+    return error
+
+
+DIR_SUPPORTED_PURLS = [
+    "pkg:generic/busybox@.*",
```

Comment:
IMHO we should later refine the logic and possibly have one concrete class for each of these base PURLs entries where we can hardcode the missing metadata, like homepage, description and so on, that we cannot get otherwise. Let's track this separately in an issue.

Reply:
We already have concrete class for these PURLs in `fetchcode` https://github.com/nexB/fetchcode/blob/b3b2052cdd9a19e0032504f6c39a943ee7cf893d/src/fetchcode/package.py#L421-L563. We can add more metadata there. Will add an issue for this in `fetchcode`.

Reply:
We're tracking this here https://github.com/nexB/fetchcode/issues/111